src/test/java/jenkins/branch/buildstrategies/basic/SkipInitialBuildOnFirstBranchIndexingTest.java

```diff
@@ -0,0 +1,46 @@
+package jenkins.branch.buildstrategies.basic;
+
+import jenkins.scm.impl.mock.MockSCMController;
+import jenkins.scm.impl.mock.MockSCMHead;
+import jenkins.scm.impl.mock.MockSCMRevision;
+import jenkins.scm.impl.mock.MockSCMSource;
+import org.junit.Test;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.*;
+
+
```

Comment:
I would prefer something more integration testy that integrates/checks the changes in https://github.com/jenkinsci/branch-api-plugin/pull/144